Language/Functions/USB/Keyboard/keyboardBegin.adoc

```diff
@@ -20,12 +20,26 @@ When used with a Leonardo or Due board, `Keyboard.begin()` starts emulating a ke
 
 [float]
 === Syntax
-`Keyboard.begin()`
+`Keyboard.begin()` +
+`Keyboard.begin(layout)`
 
 
 [float]
 === Parameters
-None
+`layout`: the keyboard layout to use. This parameter is optional and defaults to `KeyboardLayout_en_US`.
```

Comment:
Although I realize that it is missing from many of the documents (https://github.com/arduino/reference-en/issues/15), the folks working on the reference in recent years have been making some efforts to improve the coverage of parameter types documentation, following the standardized format established by the sample reference page:
https://github.com/arduino/reference-en/blob/master/AsciiDoc_sample/Reference_Terms/AsciiDoc_Template-Single_Entity.adoc#parameters

Do you think it would be reasonable to do that here @edgar-bonet?

Reply:
I do not have a strong opinion either way.

On the one hand, I think the ongoing effort of documenting the data types is of great value. Kudos for starting that! On the other hand, I expect most users to only use the pre-defined layouts. For them, I think it would be better to consider this as an _opaque_ type, i.e. a type that should be seen as an implementation detail of the library that the user does not need to care about.

If we where to document the type, how could we explain `const uint8_t *`? “Pointer to a constant `uint8_t`”, while technically correct, does not capture two essential requirements: it should point to the first element of a 128-element array, and the array should reside in `PROGMEM`. Then, “128-element long `PROGMEM` array of `uint8_t`” would be perfectly accurate, but maybe more than what the beginner would want to be told.

And then, if someone wants to make a sketch that can select the layout at run time, they may want to know the type in order to write something like:

```c++
const uint8_t *preferred_layout;
if (...)
    preferred_layout = KeyboardLayout_en_US;
else if (...)
    preferred_layout = KeyboardLayout_fr_FR;
else
    //...
Keyboard.begin(preferred_layout);
```

As you see, I am pretty unsure about the best approach here. Would you have a suggestion?

Reply:
Thanks for taking the time to share your valued insight on this.

I am also very unsure, for the exact reasons you have described. I like the idea of always documenting parameter and return types for the sake of consistency, but in cases like this it might make the documentation significantly less approachable to beginners, especially since they won't find any information about pointers and `uint8_t` in the Arduino Language Reference (https://github.com/arduino/reference-en/issues/623).

I guess that, in the current state of the reference where many other parameter types are undocumented, it is OK to leave this one undocumented as well. The question can be revisited in a more holistic context if there is a campaign to attain full parameter type documentation coverage at some future time.

So I'll consider it fine as-is for the purposes of this PR.